### DISCUSSION TITLE
Physics gun

### DISCUSSION THREAD

**Discussion by bobbydigitales** [jrouwe/JoltPhysics#540] [Q&A] [answered]
Hey! What's the best way to implement a physics gun like in Half Life 2 (or seen in Zelda Tears of the Kingdom). Is using a constraint and a positional motor the right way to go?

**Comment by jrouwe** [comment, marked as answer]
I think they work in different ways in both games so need different implementations. ```SamplesApp::CastProbe``` has a possible implementation (search for ```mDragAnchor```). If you want rotation, you could indeed use a different constraint type and have a motor. Please do some googling, there's plenty of tutorials on physics guns out there.

**Comment by bobbydigitales** [comment]
Thanks. I'll take a look.

**Comment by jankrassnigg** [comment]
See for example our implementation which uses a 6-DOF joint with motors:

https://github.com/ezEngine/ezEngine/blob/01da71c2c09863d520d7336f16dd254c816db478/Code/EnginePlugins/JoltPlugin/Constraints/Implementation/JoltGrabObjectComponent.cpp#L388
